packages/story-editor/src/components/library/panes/media/local/hotlink/hotlinkModal.js

```diff
@@ -0,0 +1,197 @@
+`;
+
+function HotlinkModal({ isOpen, onClose }) {
+  const {
+    allowedFileTypes,
+    capabilities: { hasUploadMediaAction },
+  } = useConfig();
+  const { insertElement } = useLibrary((state) => ({
+    insertElement: state.actions.insertElement,
+  }));
+
+  const { uploadVideoPoster } = useUploadVideoFrame({});
+
+  const [errorMsg, setErrorMsg] = useState(false);
+
+  // @todo We're not really uploading anything here, so should we have a fixed list instead?
+  let description = __('No file types are currently supported.', 'web-stories');
+  if (allowedFileTypes.length) {
+    description = sprintf(
+      /* translators: %s is a list of allowed file extensions. */
+      __('You can insert %s.', 'web-stories'),
+      translateToExclusiveList(allowedFileTypes)
+    );
+  }
+  const error = sprintf(
+    /* translators: %s is the description with allowed file extensions. */
+    __('Invalid link. %s', 'web-stories'),
+    description
+  );
+  const [link, setLink] = useState('');
+
+  const getFileInfo = useCallback(
+    (value = link) => {
+      const ext = value.split(/[#?]/)[0].split('.').pop().trim();
+      if (!allowedFileTypes.includes(ext)) {
+        setErrorMsg(error);
+      } else {
+        setErrorMsg(null);
+      }
+      return {
+        ext,
+        type: ['m4v', 'mp4'].includes(ext) ? 'video' : 'image',
+      };
+    },
+    [link, allowedFileTypes, error]
+  );
+
+  const onInsert = useCallback(async () => {
+    if (errorMsg?.length) {
+      return;
+    }
+    const insertionError = __(
+      'Media failed to load. Please ensure the link is valid and the site allows linking from external sites',
+      'web-stories'
+    );
+    if (!isValidUrl(link)) {
+      setErrorMsg(insertionError);
+      return;
+    }
+    try {
+      const { type } = getFileInfo();
+      const isVideo = type === 'video';
+      const getMediaDimensions = isVideo
+        ? getVideoDimensions
+        : getImageDimensions;
+      const { width, height } = await getMediaDimensions(link);
+
+      // Add necessary data for video.
+      let posterData;
+      const videoData = {};
+      if (isVideo) {
+        // Create poster if possible.
+        if (hasUploadMediaAction) {
+          const originalFileName = getFileNameFromUrl(link);
+          const fileName = getPosterName(originalFileName);
+          const posterFile = await getFirstFrameOfVideo(link);
+          posterData = await uploadVideoPoster(0, fileName, posterFile);
+          videoData.poster = posterData.poster;
+          videoData.posterId = posterData.posterId;
+        }
+        const hasAudio = await hasVideoGotAudio(link);
+        videoData.isMuted = !hasAudio;
+        const { length, formattedLength } = getVideoLength(link);
+        videoData.length = length;
+        videoData.formattedLength = formattedLength;
+      }
+      insertElement(type, {
+        resource: createResource({
+          alt: link.substring(link.lastIndexOf('/') + 1),
```

Comment:
```suggestion
          alt: link.substring(link.lastIndexOf('/') + 1),
          type,
```

Adding `type` here will fix the drop targets (now you cannot drop it as a source).

Reply:
We should also create a dedicated getResourceFromUrl helper for this, like we already have a few. createResource is a bit too low level.

Reply:
Good catch.

@swissspidy Not sure if I'll get to the helper today, I could follow up on this early next week, perhaps in a separate small issue if this is ok to merge otherwise.

Reply:
@swissspidy Creating the resource from URL is separated now, PTAL (45bae4f).